clusterupgrade/kubeClient.go

```diff
@@ -441,9 +509,33 @@ func (c KubeClient) UpdateAgentDeploymentImageVersion(namespace string, deployme
 		}
 		parts[len(parts)-1] = newImageVersion
 		newAgentImage := strings.Join(parts, ":")
-		glog.V(3).Infof(cuwlog(fmt.Sprintf("NewAgentImage is %v, will update agent deployment with this value", newAgentImage)))
-		currentDeployment.Spec.Template.Spec.Containers[0].Image = newAgentImage
 
+		glog.V(3).Infof(cuwlog(fmt.Sprintf("newImageRegistryHost is %v, will update agent deployment with this value", newImageRegistryHost)))
+		if newImageRegistryHost != "" {
+			// the deployment is using remote image registry
+			// update env
+			envs := currentDeployment.Spec.Template.Spec.Containers[0].Env
+			newEnvs := make([]v1.EnvVar, 0)
+			found := false
+			for i := range envs {
+				envar := envs[i]
+				if envar.Name == AGENT_CLUSTER_IMAGE_REGISTRY_HOST {
+					envar.Value = newImageRegistryHost
+					found = true
+				}
+				newEnvs = append(newEnvs, envar)
+			}
+			if !found {
+				newEnv := v1.EnvVar{Name: AGENT_CLUSTER_IMAGE_REGISTRY_HOST, Value: newImageRegistryHost}
+				newEnvs = append(newEnvs, newEnv)
+			}
+			currentDeployment.Spec.Template.Spec.Containers[0].Env = newEnvs
+
+			// update agent image
+			newAgentImage = fmt.Sprintf("%v/amd64_anax_k8s:%v", newImageRegistryHost, newImageVersion)
```

Comment:
is the image always going to be `amd64_anax_k8s`? 
If yes, please extract (or reuse if already exists) it to a const variable.
If no, we should construct it from existing image string.